src/CMakeLists.txt

```diff
@@ -5,3 +5,189 @@
+add_library(bitcoin_common STATIC EXCLUDE_FROM_ALL
+  base58.cpp
+  bech32.cpp
+  chainparams.cpp
+  coins.cpp
+  common/bloom.cpp
+  common/interfaces.cpp
+  common/run_command.cpp
+  $<$<TARGET_EXISTS:PkgConfig::libevent>:common/url.cpp>
+  compressor.cpp
+  core_read.cpp
+  core_write.cpp
+  deploymentinfo.cpp
+  external_signer.cpp
+  init/common.cpp
+  key.cpp
+  key_io.cpp
+  merkleblock.cpp
+  net_types.cpp
+  netaddress.cpp
+  netbase.cpp
+  net_permissions.cpp
+  outputtype.cpp
+  policy/feerate.cpp
+  policy/policy.cpp
+  protocol.cpp
+  psbt.cpp
+  rpc/rawtransaction_util.cpp
+  rpc/request.cpp
+  rpc/external_signer.cpp
+  rpc/util.cpp
+  scheduler.cpp
+  script/descriptor.cpp
+  script/miniscript.cpp
+  script/sign.cpp
+  script/signingprovider.cpp
+  script/standard.cpp
+  warnings.cpp
+)
+target_link_libraries(bitcoin_common
+  PRIVATE
+    bitcoin_consensus
+    bitcoin_util
+    univalue
+    secp256k1
+    Boost::boost
+    Threads::Threads
+    $<TARGET_NAME_IF_EXISTS:PkgConfig::libevent>
+)
+
+
+# P2P and RPC server functionality used by `bitcoind` and `bitcoin-qt` executables.
+add_library(bitcoin_node STATIC EXCLUDE_FROM_ALL
+  addrdb.cpp
+  addrman.cpp
+  banman.cpp
+  blockencodings.cpp
+  blockfilter.cpp
+  chain.cpp
+  consensus/tx_verify.cpp
+  dbwrapper.cpp
+  deploymentstatus.cpp
+  flatfile.cpp
+  headerssync.cpp
+  httprpc.cpp
+  httpserver.cpp
+  i2p.cpp
+  index/base.cpp
+  index/blockfilterindex.cpp
+  index/coinstatsindex.cpp
+  index/txindex.cpp
+  init.cpp
+  kernel/chain.cpp
+  kernel/checks.cpp
+  kernel/coinstats.cpp
+  kernel/context.cpp
+  kernel/cs_main.cpp
+  kernel/mempool_persist.cpp
+  mapport.cpp
+  net.cpp
+  netgroup.cpp
+  net_processing.cpp
+  node/blockstorage.cpp
+  node/caches.cpp
+  node/chainstate.cpp
+  node/chainstatemanager_args.cpp
+  node/coin.cpp
+  node/connection_types.cpp
+  node/context.cpp
+  node/eviction.cpp
+  node/interface_ui.cpp
+  node/interfaces.cpp
+  node/mempool_args.cpp
+  node/mempool_persist_args.cpp
+  node/miner.cpp
+  node/minisketchwrapper.cpp
+  node/psbt.cpp
+  node/transaction.cpp
+  node/txreconciliation.cpp
+  node/utxo_snapshot.cpp
+  node/validation_cache_args.cpp
+  noui.cpp
+  policy/fees.cpp
+  policy/fees_args.cpp
+  policy/packages.cpp
+  policy/rbf.cpp
+  policy/settings.cpp
+  pow.cpp
+  rest.cpp
+  rpc/blockchain.cpp
+  rpc/fees.cpp
+  rpc/mempool.cpp
+  rpc/mining.cpp
+  rpc/net.cpp
+  rpc/node.cpp
+  rpc/output_script.cpp
+  rpc/rawtransaction.cpp
+  rpc/server.cpp
+  rpc/server_util.cpp
+  rpc/signmessage.cpp
+  rpc/txoutproof.cpp
+  script/sigcache.cpp
+  shutdown.cpp
+  signet.cpp
+  timedata.cpp
+  torcontrol.cpp
+  txdb.cpp
+  txmempool.cpp
+  txorphanage.cpp
+  txrequest.cpp
+  validation.cpp
+  validationinterface.cpp
+  versionbits.cpp
+
+  dummywallet.cpp
+)
+target_link_libraries(bitcoin_node
+  PRIVATE
+    bitcoin_common
+    bitcoin_util
+    leveldb
+    minisketch
+    univalue
+    Boost::boost
```

Comment:
Why is Boost in here? We don't link any Boost libs.

Reply:
`Boost::boost` is an imported "library" which actually provides only included headers as [usage requirements](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-specification-and-usage-requirements).

Reply:
How nice and confusing. This will need some commentary. I assume there's no way for this to accidently start linking against actual Boost libraries going forward?

Reply:
> This will need some commentary.

Is that enough: https://github.com/hebasto/bitcoin/blob/61aa6e20bee512c9d0ee2845380dbb67af1c5d43/CMakeLists.txt#L95-L96

? Or what comment would be more appropriate?

> I assume there's no way for this to accidently start linking against actual Boost libraries going forward?

Yes, I think so. Actual Boost libraries are associated with targets like [`Boost::<component>`](https://cmake.org/cmake/help/latest/module/FindBoost.html#imported-targets).

Reply:
We should probably mention that this can be changed to `boost::headers` once we are using 3.15+. Is there a way for us to create/set that as some sort of alias in the interim?

Reply:
> We should probably mention that this can be changed to `boost::headers` once we are using 3.15+. Is there a way for us to create/set that as some sort of alias in the interim?

Yes. Going to implement your suggestion.

Reply:
> We should probably mention that this can be changed to `boost::headers` once we are using 3.15+. Is there a way for us to create/set that as some sort of alias in the interim?

Thanks! [Updated](https://github.com/hebasto/bitcoin/pull/7#issuecomment-1443742611).

Reply:
I hadn't yet seen this discussion when I commented [here](https://github.com/hebasto/bitcoin/pull/7#discussion_r1117585607). I think this is a very nice and clever compromise. Thanks @fanquake.

Reply:
[Reworked](https://github.com/hebasto/bitcoin/pull/7#issuecomment-1445212086) and a comment has been added.